drivers/cc2420/cc2420.c

```diff
@@ -172,6 +172,10 @@ int cc2420_rx(cc2420_t *dev, uint8_t *buf, size_t max_len, void *info)
         cc2420_ram_read(dev, CC2420_RAM_RXFIFO, &len, 1);
         len -= 2;   /* subtract RSSI and FCF */
         DEBUG("cc2420: recv: packet of length %i in RX FIFO\n", (int)len);
+        if (max_len != 0) {
+            /* dropping frame was requested */
+            cc2420_strobe(dev, CC2420_STROBE_FLUSHRX);
```

Comment:
AFAIS since the second `cc2420_strobe` is in the `else` block, this function will only get called once. Thus, the FIFO won't be flushed properly.

I guess that could be solved putting both `cc2420_strobe` in the end and adding `return` statement with the length here.